src/EnergyPlus/ResultsFramework.cc

```diff
@@ -973,344 +927,105 @@ namespace ResultsFramework {
         return outputMsgPack;
     }
 
-    void ResultsFramework::initializeRTSDataFrame(const OutputProcessor::ReportingFrequency reportFrequency,
-                                                  const Array1D<RealVariableType> &RVariableTypes,
-                                                  const int NumOfRVariable,
-                                                  const OutputProcessor::TimeStepType timeStepType)
-    {
-        Reference<RealVariables> RVar;
-
-        for (int Loop = 1; Loop <= NumOfRVariable; ++Loop) {
-            RVar >>= RVariableTypes(Loop).VarPtr;
-            auto &rVar = RVar();
-            if (rVar.Report && rVar.frequency == reportFrequency) {
-                Variable var;
-                if (RVariableTypes(Loop).units == OutputProcessor::Unit::customEMS) {
-                    var = Variable(RVariableTypes(Loop).VarName,
-                                   reportFrequency,
-                                   RVariableTypes(Loop).timeStepType,
-                                   RVariableTypes(Loop).ReportID,
-                                   RVariableTypes(Loop).units,
-                                   RVariableTypes(Loop).unitNameCustomEMS);
+    void ResultsFramework::initializeTSDataFrame(const OutputProcessor::ReportFreq reportFrequency,
+                                                 const std::vector<OutputProcessor::OutVar *> &Variables,
+                                                 const OutputProcessor::TimeStepType timeStepType)
+    {
+        for (auto *var : Variables) {
+            
+            if (var->Report && var->freq == reportFrequency) {
+                Variable rfvar;
+                if (var->units == Constant::Units::customEMS) {
+                    rfvar = Variable(var->keyColonName, reportFrequency, var->timeStepType, var->ReportID, var->units, var->unitNameCustomEMS);
                 } else {
-                    var = Variable(RVariableTypes(Loop).VarName,
-                                   reportFrequency,
-                                   RVariableTypes(Loop).timeStepType,
-                                   RVariableTypes(Loop).ReportID,
-                                   RVariableTypes(Loop).units);
+                    rfvar = Variable(var->keyColonName, reportFrequency, var->timeStepType, var->ReportID, var->units);
                 }
+                
                 switch (reportFrequency) {
-                case OutputProcessor::ReportingFrequency::EachCall: // each time UpdatedataandReport is called
```

Comment:
These are kept in arrays now, so just index by reportFreq rather than replicate code.